src/toaster/src/Toast.js

```diff
@@ -47,154 +47,153 @@ const animationStyles = css({
   }
 })
 
-export default class Toast extends React.PureComponent {
-  static propTypes = {
-    /**
-     * The z-index of the toast.
-     */
-    zIndex: PropTypes.number,
-
-    /**
-     * Duration of the toast.
-     */
-    duration: PropTypes.number,
-
-    /**
-     * Function called when the toast is all the way closed.
-     */
-    onRemove: PropTypes.func,
-
-    /**
-     * The type of the alert.
-     */
-    intent: PropTypes.oneOf(['none', 'success', 'warning', 'danger'])
-      .isRequired,
-
-    /**
-     * The title of the alert.
-     */
-    title: PropTypes.node,
-
-    /**
-     * Description of the alert.
-     */
-    children: PropTypes.node,
-
-    /**
-     * When true, show a close icon button inside of the toast.
-     */
-    hasCloseButton: PropTypes.bool,
-
-    /**
-     * When false, will close the Toast and call onRemove when finished.
-     */
-    isShown: PropTypes.bool
-  }
-
-  static defaultProps = {
-    intent: 'none'
-  }
-
-  state = {
-    isShown: true,
-    height: 0
-  }
-
-  componentDidUpdate(prevProps) {
-    if (prevProps.isShown !== this.props.isShown) {
-      // eslint-disable-next-line react/no-did-update-set-state
-      this.setState({
-        isShown: this.props.isShown
-      })
+const Toast = memo((props) => {
+  const {
+    duration,
+    onRemove,
+    isShown: isShownProp,
+    // Template props
+    intent = 'none',
+    zIndex,
+    title,
+    children,
+    hasCloseButton
+  } = props
+
+  const [isShown, setIsShown] = useState(true)
+  const [closeTimer, setCloseTimer] = useState(null)
+  const [height, setHeight] = useState(0)
+
+  const close = () => {
+    clearCloseTimer()
```

Comment:
Shouldn't `clearCloseTimer` be defined first?

Reply:
Yes good catch